### ISSUE TITLE
v5 support

### ISSUE THREAD

**Issue by VetSoftKeith** [mb3364/Java-Twitch-Api-Wrapper#22]
With Twitch releasing their v5 API last week, are you planning on doing an update to accommodate v5 once the docs are released?

**Comment by BobbySig** [comment]
@keesh7663 I'm considering doing the upgrade myself, but my issue is that there are some pretty major changes on the `develop` branch that I don't know the status of. @urgrue can anything be done with those changes or should I just ignore them and build off of the current `master` branch?

**Comment by Ollee** [comment]
Has anyone heard anything from urgrue?  @BobbySig ?

**Comment by BobbySig** [comment]
I haven't heard anything from @urgrue

**Comment by Gikkman** [comment]
I looked through the changes in the `development` branch, at it mostly looks like refactoring and preparations to move over to another Http library. Hence, I don't think they are to big of a loss to ditch in case someone wants to extend this library.

**Comment by VetSoftKeith** [comment]
I've started the upgrade to @urgue wrapper code...  I should have a solid foundation and a repo coming somewhat soon

**Comment by BobbySig** [comment]
@Gikkman true, though making the swap to the new Async Http Client (the successor to the project @urgrue was switching to) wouldn't be a bad idea, as that project is under active development and seems to work very well.

@keesh7663 awesome!

**Comment by urgrue** [comment]
Hey everyone, sorry for my silence! I do intend to update the library for v5 support, but won't have any time until mid- next month. I'll gladly accept pull requests if someone beats me to it.

Regarding the `develop` branch: As you all noticed I had started refactoring to move to a more robust and tested http client library. My ultimate goal is to provide an interface that this library will use, provide a default implementation, and also allow one to implement that same interface with any http library of your choosing. This way if your project is already using a different http library, you aren't forced into taking on the extra dependency of the additional http library.

**Comment by Gikkman** [comment]
Yeah, that move sounds like a good idea @urgrue Nice to see you back, though I hope you didn't return because you felt forced too. 

Will you keep on working with this, @keesh7663, or do you intend to wait and let urgrue have a go at it first? What would you prefer urgue?

**Comment by VetSoftKeith** [comment]
well, it's been slow going on my end due to other commitments, but if @urgrue has it under control then I can stop the work on it...  i probably wouldn't have a final working copy of it until about the same sort of time-frame

**Comment by VetSoftKeith** [comment]
Just wanted to update here:  I've got all the endpoints that currently exist in the v3 version of the wrapper done.  I just need to add the 2 new endpoints (which I will be working on over the next few days).  Another thing I am looking at doing is switching it over to using the Apache Http Client at some point.

@Gikkman @urgrue @BobbySig

**Comment by Gikkman** [comment]
That's great to hear. 

I also agree that it might be a good idea to switch to a more stock Http library. The ideal solution would probably be to somehow make it modular (e.g. users can chose which Http client they want to use). At the same time, in order to keep the library accessible, it is probably better to bundle a Http library with it. 

Again, great to hear that you are making steady progress.

**Comment by VetSoftKeith** [comment]
@Gikkman if you want to help test it out, you can find it [here](https://github.com/VeteranSoftware/Java-Twitch-Api-Wrapper)

**Comment by Gikkman** [comment]
Sure, I'll take a look

**Comment by VetSoftKeith** [comment]
@urgue so I haven't been able to finish the rewrite.  RL kinda got in the way of that

**Comment by kolatra** [comment]
Anything new on this?

**Comment by MaximKulikov** [comment]
@keesh7663 Could you add issue tracker to you fork? There is some mess with Follow model. Or i can fix it myself and make pull request.

**Comment by VetSoftKeith** [comment]
Version 5 of the Twitch API is now deprecated and scheduled for removal in 1 year.  so i won't be working on that particular fork anymore.
